force-app/main/default/classes/DMLOperations.cls

```diff
@@ -211,9 +344,29 @@ public with sharing class DMLOperations {
      * @param accountId The account for which the cases will be created and then deleted.
      * @param numOfCases The number of cases to create.
      */
-    public static void createAndDeleteCases(String accountId, Integer numOfCases) {
+    public static void createAndDeleteCases(String accountId, Integer numOfCases) 
+    {
         // Insert the cases
+        List<Case> casesToCreate = new List<Case>();
+
+    for (Integer i = 0; i < numOfCases; i++) 
+    {
+        Case newCase = new Case(
+            AccountId = accountId,
+            Subject = 'Auto-generated Case ' + (i + 1),
+            Status = 'New'
+        );
+        casesToCreate.add(newCase);
+    }
 
+    if (!casesToCreate.isEmpty()) 
+    {
+        insert casesToCreate;
+    }
         // Delete the cases
+        if (!casesToCreate.isEmpty()) 
+        {
+            delete casesToCreate;
+        }
```

Comment:
Why have 2 if checks here, why not just the one? Also, some people will say it is not necessary to check is empty before DML, the insert and delete will check for you and not use DML limits if there is nothing in the list.